apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -86,26 +115,67 @@ const CreateOutOfOfficeEntryModal = ({
     },
   });
 
-  const createOutOfOfficeEntry = trpc.viewer.outOfOfficeCreate.useMutation({
+  const resetForm = useCallback(() => {
+    reset();
+    setSelectedReason(null);
+    setSelectedMember(null);
+    setProfileRedirect(false);
+  }, [reset]);
+
+  const createOrEditOutOfOfficeEntry = trpc.viewer.outOfOfficeCreateOrUpdate.useMutation({
     onSuccess: () => {
-      showToast(t("success_entry_created"), "success");
+      showToast(
+        currentlyEditingOutOfOfficeEntry
+          ? t("success_edited_entry_out_of_office")
+          : t("success_entry_created"),
+        "success"
+      );
       utils.viewer.outOfOfficeEntriesList.invalidate();
-      setProfileRedirect(false);
+      resetForm();
       closeModal();
     },
     onError: (error) => {
       showToast(t(error.message), "error");
     },
   });
 
-  const { data: outOfOfficeReasonList } = trpc.viewer.outOfOfficeReasonList.useQuery();
+  useEffect(() => {
+    if (currentlyEditingOutOfOfficeEntry === null) {
+      return;
+    }
 
-  const reasonList = [
-    ...(outOfOfficeReasonList || []).map((reason) => ({
-      label: `${reason.emoji} ${reason.userId === null ? t(reason.reason) : reason.reason}`,
-      value: reason.id,
-    })),
-  ];
+    const selectedReason =
+      reasonList.find((reason) => {
+        return reason.value === currentlyEditingOutOfOfficeEntry.selectedReason;
+      }) ?? null;
+    if (selectedReason && selectedReason.value) {
+      setValue("reasonId", selectedReason.value);
+      setSelectedReason(selectedReason);
+    }
+
+    const selectedMember =
+      memberListOptions.find((member) => {
+        return member.value === currentlyEditingOutOfOfficeEntry.selectedMember;
+      }) ?? null;
+
+    setProfileRedirect(currentlyEditingOutOfOfficeEntry.profileRedirect);
+
+    if (selectedMember && selectedMember?.value) {
+      setValue("toTeamUserId", selectedMember.value);
+      setSelectedMember(selectedMember);
+    }
+
+    if (currentlyEditingOutOfOfficeEntry.notes) {
+      setValue("notes", currentlyEditingOutOfOfficeEntry.notes);
+    }
+
+    setValue("dateRange", {
+      startDate: dayjs(currentlyEditingOutOfOfficeEntry.start).startOf("d").toDate(),
+      endDate: dayjs(currentlyEditingOutOfOfficeEntry.end).subtract(1, "d").endOf("d").toDate(),
+    });
+
+    setValue("uuid", currentlyEditingOutOfOfficeEntry.uuid);
+  }, [currentlyEditingOutOfOfficeEntry, setValue, reasonList, dateRange, memberListOptions]);
```

Comment:
Added an example before which conditionally sets defaultValue, so this won't be needed anymore